### PR TITLE
Change graphic order with context menu

### LINKED ISSUE #655: Move image back and forward

### PR DESCRIPTION
**Pull request by bigfoodK**
- Add graphic order changing buttons
  - Send To Back
  - Send Backward
  - Bring Forward
  - Bring To Front
- Fix indexing bug
  - Indexing in `Cut/SequenceUpdateAction::MoveCut/ChangeGraphicOrder` was wrong. and dragging was made for that indexing.
    - Fix indexings and draggings too

image

Closes #655

### CODE REVIEW

**Comment by namse** [comment]
I hope binding the same key with google slides (ctrl+page up or down I guess?)

File: luda-editor/client/src/pages/sequence_edit_page/components/wysiwyg_editor/mod.rs [hunk 1]
```diff
@@ -208,6 +208,85 @@ impl Component for WysiwygEditor<'_> {
...
+                        .add_button("Send Backward", || {
+                            let Some(next_or_last_graphic_index) = ({
+                                let Some(position) = screen_graphics
+                                    .iter()
+                                    .position(|(index, _)| *index == graphic_index)
+                                else {
+                                    return;
+                                };
+                                let next_position = (position + 1).min(screen_graphics.len() - 1);
+                                screen_graphics.get(next_position).map(|(index, _)| *index)
```
Comment (namse) [inline luda-editor/client/src/pages/sequence_edit_page/components/wysiwyg_editor/mod.rs:237]: Personally looks so ugly :( using `map` fn must be better

**Review by namse — CHANGES_REQUESTED** [review]

File: luda-editor/client/src/pages/sequence_edit_page/components/wysiwyg_editor/mod.rs (hunk 1 above)
Comment (bigfoodK) [reply luda-editor/client/src/pages/sequence_edit_page/components/wysiwyg_editor/mod.rs:237]:
```rust
screen_graphics
    .iter()
    .position(|(index, _)| *index == graphic_index)
    .map(|position| {
        let next_position =
            (position + 1).min(screen_graphics.len() - 1);
        screen_graphics.get(next_position).map(|(index, _)| *index)
    })
```

I've been trying to reduce depth recently.
But, yes, it looks much better.

Comment (bigfoodK) [reply luda-editor/client/src/pages/sequence_edit_page/components/wysiwyg_editor/mod.rs]: Fixed with [291a7eb](https://github.com/NamseEnt/namseent/pull/681/commits/291a7eb804658b3235e473217a04dc5dffab2644)

**Comment by namse** [comment]
> I hope binding the same key with google slides (ctrl+page up or down I guess?)

Add new issue or handle in this pr? @bigfoodK

**Comment by bigfoodK** [comment]
> > I hope binding the same key with google slides (ctrl+page up or down I guess?)
> 
> Add new issue or handle in this pr? @bigfoodK

Yes, I opened #682

**Review by namse — APPROVED** [review]

**Pull request merged**